### PR TITLE
VB-3435 - Orchestration endpoints for get prisoner and prisoners visitor details.

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PrisonerProfileController.kt [hunk 1]
```diff
@@ -53,4 +56,40 @@ class PrisonerProfileController(
   fun getPrisonerProfile(@PathVariable prisonId: String, @PathVariable prisonerId: String): PrisonerProfileDto? {
     return prisonerProfileService.getPrisonerProfile(prisonId, prisonerId)
   }
+
+  @PreAuthorize("hasRole('ROLE_ORCHESTRATION_SERVICE__VISIT_BOOKER_REGISTRY')")
```
Comment (tpmcgowan) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PrisonerProfileController.kt:60]: Is this the role the Public UI will need to use for this endpoint? If so, I _think_ it should be `ROLE_VISIT_ORCHESTRATION__PUBLIC_VISIT_BOOKING_UI` to match the proposed convention 🤔. Perhaps worth a chat...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PrisonerProfileController.kt [hunk 2]
```diff
@@ -10,16 +10,19 @@ import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RestController
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.config.ErrorResponse
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.PrisonerProfileDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.prisoner.search.PrisonerBasicInfoDto
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.service.PrisonerProfileService
 
 const val ORCHESTRATION_PRISONER_CONTROLLER_PATH: String = "/prisoner"
+const val ORCHESTRATION_PRISONER_PROFILE_CONTROLLER_PATH: String = "$ORCHESTRATION_PRISONER_CONTROLLER_PATH/{prisonId}/{prisonerId}/profile"
+const val ORCHESTRATION_PRISONERS_BASIC_INFO_CONTROLLER_PATH: String = "$ORCHESTRATION_PRISONER_CONTROLLER_PATH/{prisonerIds}/basic-details"
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PrisonerProfileController.kt:19]: not sure about the nameing,... notmally w would have prisoners/3233223/prison/CFI something like that,..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt [hunk 3]
```diff
@@ -69,4 +87,8 @@ class PrisonerProfileService(
       size = MAX_VISIT_RECORDS,
     )
   }
+
+  private fun getBlankPrisonerBasicInfo(prisonerId: String): PrisonerBasicInfoDto {
+    return PrisonerBasicInfoDto(prisonerId, null, null, null, null)
+  }
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt:93]: don't understand this,... if it has not got the details we should not return them I think?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitorProfileService.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        BasicContactDto(it)
+      } ?: getBlankContactInfo(visitorId)
+
+      contacts.add(contact)
+    }
+    return contacts
+  }
+
+  private fun getBlankContactInfo(visitorId: Long): BasicContactDto {
+    return BasicContactDto(visitorId, null, null)
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitorProfileService.kt:32]: same goes for this also

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/prisoner/search/PrisonerBasicInfoDto.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import io.swagger.v3.oas.annotations.media.Schema
+import java.time.LocalDate
+
+data class PrisonerBasicInfoDto(
+  @Schema(description = "Prisoner Number", example = "A1234AA", required = true)
+  val prisonerNumber: String,
+
+  @Schema(description = "First Name", example = "Robert", required = false)
+  val firstName: String?,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/prisoner/search/PrisonerBasicInfoDto.kt:11]: these should not be optional

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/contact/registry/BasicContactDto.kt [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import io.swagger.v3.oas.annotations.media.Schema
+
+@Schema(description = "A contact for a prisoner")
+class BasicContactDto(
+  @Schema(description = "Identifier for this contact (Person in NOMIS)", example = "5871791", required = true)
+  val personId: Long,
+  @Schema(description = "First name", example = "John", required = false)
+  val firstName: String?,
+  @Schema(description = "Last name", example = "Smith", required = false)
+  val lastName: String?,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/contact/registry/BasicContactDto.kt:12]: these should not be optional

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt (hunk 3 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt]: updated to not return if not found.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitorProfileService.kt (hunk 4 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitorProfileService.kt]: updated to not return if not found.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/prisoner/search/PrisonerBasicInfoDto.kt (hunk 5 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/prisoner/search/PrisonerBasicInfoDto.kt]: Updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/contact/registry/BasicContactDto.kt (hunk 6 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/dto/contact/registry/BasicContactDto.kt]: Updated.

**Review by AledWEvans — APPROVED** [review]
LGTM

**Review by AledWEvans — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PrisonerProfileController.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PrisonerProfileController.kt:19]: https://dsdmoj.atlassian.net/browse/VB-3734 raised to correct existing prisoner profile endpoint path

**Pull request merged**